Assets/Scripts/Destructible.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/***********************************************************
+ * Script will change an object's texture to appear damage
+ * Accomplished by changing Sprites when object is within a
+ * certain range (Health calculated in percent)
+ ***********************************************************/
+public class Destructible : Entity
+{
+    public Sprite[] Sprites; //The different textures go into this array
+    private float[] HealthPercents;
+    private float MaxHealth; //Constant value; an objects maximum health
+    private float prevHealth;
+
+    //Private method determines the cutoffs on percent the ranges based on the number of Sprites
+    private float[] CreatePercents(int numSprites)
+    {
+        float[] HealthPercents = new float[numSprites];
+
+        HealthPercents[0] = 100;
+        if (numSprites > 1)
+        {
+            float DISTANCE = 100 / (numSprites - 1); //Constant value; Distance from one cutoff to another
+            for (int i = 1; i < numSprites; ++i)
+            {
+                HealthPercents[i] = HealthPercents[i - 1] - DISTANCE;
+            }
+        }
+        return HealthPercents;
+    }
+
+    //Private method which will gauge what texture to display when a health percentage is within a desired range
+    private void ChangeTextures()
+    {
+        float currPercent;
+
+        for (int i = 0; i < HealthPercents.Length - 1; ++i)
+        {
+            currPercent = base.health / MaxHealth * 100;
+
+            if (currPercent < HealthPercents[i] & currPercent >= HealthPercents[i + 1])
+            {
+                gameObject.GetComponent<SpriteRenderer>().sprite = Sprites[i + 1];
```

Comment:
So I'm not sure what exactly is happening here. Can you explain to me the logic here?

Please correct me if I'm wrong, but it seems like you are potentially making a new sprite for each health point?

Reply:
So the HealthPercents array basically highlights the cutoffs at which the object should change textures (Array would typically look like: 100, 75, 50, 25, 0). If currPercent, which is the object's current health as a percentage, is within a certain range (Like between 75 - 50), then it will change to the next sprite. Else, it will just stay as as its current Sprite texture.

Reply:
Ah I see, so these are like different damage sprites? So as the Destructable gets more damaged?